fix(Leave Application): calendar view

barredterra opened the pull request:
- Fix date range condition

	```
	Leave starts before/on last day of current view
	AND Leave ends after/on first day of current view
	```

- Convert start and end parameters from _datetime_ to _date_ before passing them to the query
- Fix property name `"all_day`" -> `"allDay"` as expected by frontend
- Show holidays as "allDay"
    - Regarding half day, it doesn't make sense to mark an entire leave of, say, 5 days as "not all day" or "half day", just because the first or the last of the 5 days is a half day leave.
- Fix display of blocked days
    - Show as "allDay"
    - Remove unused whitelisted method
    - Query applicable blocked days in the correct order

        (If `company` is available and `employee` is not, we should still return the block lists that apply to all departments.)

Please backport to v14 + v15

barredterra committed:
fix(Leave Application): calendar view

barredterra committed:
fix: show blocked dates as "allDay"

barredterra committed:
fix: remove unused whitelisting from get_applicable_block_dates

barredterra committed:
fix: get blocked days in the correct order

If company is available and employee is not, we should still return the block lists that apply to all departments.

barredterra committed:
fix: convert datetime to date

barredterra committed:
Merge branch 'develop' into leave-application-calendar

krantheman commented on hrms/hr/doctype/leave_application/leave_application.py:
```diff
@@ -1315,6 +1321,7 @@ def add_holidays(events, start, end, employee, company):
 				"to_date": holiday.holiday_date,
 				"title": _("Holiday") + ": " + cstr(holiday.description),
 				"name": holiday.name,
+				"allDay": True,
```
```suggestion
				"allDay": 1,
```

 This can be set to 1 for consistency

krantheman commented on hrms/hr/doctype/leave_block_list/leave_block_list.py:
```diff
@@ -59,33 +58,34 @@ def get_applicable_block_dates(
+		add_block_list(frappe.db.get_all("Leave Block List", filters=conditions, pluck="name"))
+
+	if not employee:
+		employee = frappe.db.get_value("Employee", {"user_id": frappe.session.user})
```
Since employee is used to fetch company, should this be added on top?

krantheman reviewed (dismissed):
Some of the tests are failing, can you look into those?

barredterra committed:
Merge branch 'develop' into leave-application-calendar

barredterra committed:
refactor: use `1` instead of `True`

barredterra committed:
fix: set employee before company

barredterra committed:
fix: half day

I doesn't make sense to mark the entire leave of, say, 5 days as "not all day" or "half day", just because the first or the last of the 5 days is a half day leave.

barredterra commented:
@krantheman tests pass now. The linter failure is unrelated (I won't rewrite all the existing SQL queries in this PR).

krantheman committed:
refactor: use get_list instead of sql and simplify date range condition

krantheman committed:
fix: remove employee_name and leave_type before checking in events

krantheman committed:
fix: use get_single_value instead of get_value

krantheman committed:
chore: add comment
